main.go

```diff
@@ -559,56 +559,61 @@ func Reload(logger *util.Logger) {
 	os.Exit(0)
 }
 
-func doLogTest(servers []*state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) {
+func doLogTest(servers []*state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) bool {
 	// Initial test
 	hasFailedServers, hasSuccessfulLocalServers := runner.TestLogsForAllServers(servers, globalCollectionOpts, logger)
 
 	// Re-test using lower privileges
 	if hasFailedServers {
-		return
+		return false
+	}
+	if !hasSuccessfulLocalServers {
+		return true
 	}
-	if hasSuccessfulLocalServers {
-		curUser, err := user.Current()
-		if err != nil {
-			logger.PrintError("Could not determine current user for privilege drop test")
-			return
-		}
 
-		pgaUser, err := user.Lookup("pganalyze")
-		if err != nil {
-			logger.PrintVerbose("Could not locate pganalyze user, skipping privilege drop test: %s", err)
-			return
-		} else if curUser.Name != "root" {
-			logger.PrintVerbose("Current user is not root, skipping privilege drop test")
-			return
-		} else if curUser.Uid == pgaUser.Uid {
-			logger.PrintVerbose("Current user is already pganalyze user, skipping privilege drop test")
-			return
-		}
+	curUser, err := user.Current()
+	if err != nil {
+		logger.PrintError("Could not determine current user for privilege drop test")
+		return false
+	}
+	if curUser.Name != "root" {
+		// don't print anything here, since it would always be printed during the actual privilege drop run
+		return true
+	}
 
-		uid, _ := strconv.ParseUint(pgaUser.Uid, 10, 32)
-		gid, _ := strconv.ParseUint(pgaUser.Gid, 10, 32)
-		groupIDStrs, _ := pgaUser.GroupIds()
-		var groupIDs []uint32
-		for _, groupIDStr := range groupIDStrs {
-			groupID, _ := strconv.ParseUint(groupIDStr, 10, 32)
-			groupIDs = append(groupIDs, uint32(groupID))
-		}
-		logger.PrintInfo("Re-running log test with reduced privileges of \"pganalyze\" user (uid = %d, gid = %d)", uid, gid)
-		collectorBinaryPath, err := os.Executable()
-		if err != nil {
-			logger.PrintError("Could not run collector log test as \"pganalyze\" user due to missing executable: %s", err)
-			return
-		}
-		cmd := exec.Command(collectorBinaryPath, "--test-logs")
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-		cmd.SysProcAttr = &syscall.SysProcAttr{}
-		cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uint32(uid), Gid: uint32(gid), Groups: groupIDs}
-		err = cmd.Run()
-		if err != nil {
-			logger.PrintError("Could not run collector log test as \"pganalyze\" user: %s", err)
-			return
-		}
+	pgaUser, err := user.Lookup("pganalyze")
+	if err != nil {
+		logger.PrintVerbose("Could not locate pganalyze user, skipping privilege drop test: %s", err)
+		return false
+	} else if curUser.Uid == pgaUser.Uid {
+		logger.PrintVerbose("Current user is already pganalyze user, skipping privilege drop test")
```

Comment:
Should this be `PrintError`? Or should we return `true` in these cases? Right now the output can be confusing:

```
$ sudo ./pganalyze-collector --test-logs --reload --config=./pganalyze-collector-local.conf
2021/02/17 11:50:27 I Running collector test with pganalyze-collector 0.36.0
2021/02/17 11:50:27 I [server1] Testing log collection (local)...
2021/02/17 11:50:33 I [server1]   Local log test successful
2021/02/17 11:50:33 E Error: Reload requested, but ignoring since configuration errors are present
$ sudo ./pganalyze-collector --test-logs --reload --config=./pganalyze-collector-local.conf --verbose
2021/02/17 11:50:51 V [server1] Identified as api_system_type: self_hosted, api_system_scope: 5432/mydb1*, api_system_id: starling
2021/02/17 11:50:51 V Did not open state file: open /var/lib/pganalyze-collector/state: no such file or directory
2021/02/17 11:50:51 I Running collector test with pganalyze-collector 0.36.0
2021/02/17 11:50:51 I [server1] Testing log collection (local)...
2021/02/17 11:50:51 V [server1] Searching for log file(s) in /var/log/postgresql/postgresql-11-main.log
2021/02/17 11:50:51 V [server1] Tailing log file /var/log/postgresql/postgresql-11-main.log
2021/02/17 11:50:57 I [server1]   Local log test successful
2021/02/17 11:50:57 V [server1] Log file fsnotify watcher received stop signal
2021/02/17 11:50:57 V [server1] Stopping log tail for /var/log/postgresql/postgresql-11-main.log (stop requested)
2021/02/17 11:50:57 V Could not locate pganalyze user, skipping privilege drop test: user: unknown user pganalyze
2021/02/17 11:50:57 E Error: Reload requested, but ignoring since configuration errors are present
```
(The first test fails because I don't have a pganalyze user on this machine, but the test output does not make that clear.)

Reply:
You commented on the line that has `logger.PrintVerbose("Current user is already pganalyze user, skipping privilege drop test")` - I assume you actually meant the line that has `logger.PrintVerbose("Could not locate pganalyze user, skipping privilege drop test: %s", err)`?

I think we could consider making that a warning, and doing `return true` in that case. Setups that don't have a pganalyze user are rare (they'd be self compiled), but it might be nice to support running the collector as root in such situations (and having a successful `--test-logs` command), if someone really wants to.

I think the if branch this comment is on `( if curUser.Uid == pgaUser.Uid {`) can be left as-is.